pennylane/math/torch_box.py

```diff
@@ -117,6 +126,12 @@ def numpy(self):
     def requires_grad(self):
         return self.data.requires_grad
 
+    @wrap_output
+    def scatter_element_add(self, index, value):
+        self.data = self.data.clone()
+        self.data[tuple(index)] += value
+        return self.data

```

Comment:
So this function is implemented not in-place?

Reply:
It's a limitation of PyTorch :(


PyTorch allows in-place operations of tensors, *unless* the tensor is a leaf node (that is, a tensor created at the beginning of the computational graph by a user, with the `requires_grad` explicitly set).

But I just realised we can have a check for this using `self.data.is_leaf`, so we only need to clone if that is true.